components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -601,6 +618,183 @@ fn cast_as_string_helper(
     Ok(Some(res))
 }
 
+// cast any as decimal, some cast functions reuse `cast_any_as_decimal`
+//
+// - cast_signed_int_as_signed_decimal -> cast_any_as_decimal<Int>
+// - cast_string_as_signed_decimal -> cast_any_as_decimal<Bytes>
+// - cast_time_as_decimal -> cast_any_as_decimal<Time>
+// - cast_duration_as_decimal -> cast_any_as_decimal<Duration>
+// - cast_json_as_decimal -> cast_any_as_decimal<Json>
+
+// because uint's upper bound is smaller than signed decimal's upper bound
+// so we can merge cast uint as signed/unsigned decimal in this function
+#[rpn_fn(capture = [ctx, extra])]
+#[inline]
+fn cast_unsigned_int_as_signed_or_unsigned_decimal(
+    ctx: &mut EvalContext,
+    extra: &RpnFnCallExtra,
+    val: &Option<i64>,
+) -> Result<Option<Decimal>> {
+    match val {
+        None => Ok(None),
+        Some(val) => {
+            let dec = Decimal::from(*val as u64);
+            Ok(Some(produce_dec_with_specified_tp(
+                ctx,
+                dec,
+                extra.ret_field_type,
+            )?))
+        }
+    }
+}
+
+#[rpn_fn(capture = [ctx, extra])]
+#[inline]
+fn cast_signed_int_as_unsigned_decimal(
+    ctx: &mut EvalContext,
+    extra: &RpnFnCallExtra<'_>,
+    val: &Option<i64>,
+) -> Result<Option<Decimal>> {
+    match val {
+        None => Ok(None),
+        Some(val) => {
+            let dec = if in_union(extra.implicit_args) && *val < 0 {
+                Decimal::zero()
+            } else {
+                // FIXME: here TiDB has bug, fix this after fix TiDB's
```

Comment:
Could we have a link to that issue? And we can make a comment there so as to get notified when they are complete.

Reply:
It's better to file an issue to TiDB repo or TiKV repo to point the real problem instead of adding some comments like this (this comment does not contain any useful information).